crates/matrix-sdk/src/encryption/mod.rs

```diff
@@ -202,13 +203,23 @@ impl Client {
                 MessageType::Image(content)
             }
             mime::AUDIO => {
-                let info = assign!(info.map(AudioInfo::from).unwrap_or_default(), {
+                let audio_info = assign!(info.clone().map(AudioInfo::from).unwrap_or_default(), {
                     mimetype: Some(content_type.as_ref().to_owned()),
                 });
-                let content = assign!(AudioMessageEventContent::encrypted(body.to_owned(), file), {
-                    info: Some(Box::new(info))
+
+                let mut audio_message_event_content = assign!(AudioMessageEventContent::encrypted(body.to_owned(), file), {
+                    info: Some(Box::new(audio_info))
                 });
-                MessageType::Audio(content)
+
+                if let Some(AttachmentInfo::Voice(audio_info, Some(waveform))) = info {
+                    if let Some(duration) = audio_info.duration {
+                        let waveform = waveform.iter().map(|v| (*v).into()).collect();
```

Comment:
What's the target type? Would it make sense that the vec in `AttachmentInfo::Voice` immediately store the target type instead?

Reply:
The final result is a `AudioMessageEventContent`. I don't think it makes sense for `::Voice` to store this ruma type into it.

Reply:
discussed privately